### PR TITLE
GCI-142: Fix bad code practices through Sonar

### PR DESCRIPTION
**Pull request by collinsnji**
## Description

GCI-142: Fixed bad code practices by removing unused imports
## Related Issue

see https://issues.openmrs.org/browse/TRUNK-

## Checklist:

- [x] My pull request only contains one single commit.
- [x] My pull request is based on the latest master branch
  `git pull --rebase upstream master`.
- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by dkayiwa** [comment]
@collinsnji did you take a look at the above travis failure?

File: api/src/main/java/org/openmrs/hl7/impl/HL7ServiceImpl.java [hunk 1]
```diff
@@ -393,6 +393,18 @@ public HL7InError getHL7InErrorByUuid(String uuid) throws APIException {
...
+ 	 * @param idNum id number
+ 	 * @param fName family name
+ 	 * @param gName given name
+ 	 * @return error string. User can not be resolveUserId
+ 	 */
+ 	public String errorFindingUser(String idNum, String fName, String gName) {
+ 		String cantFindUser = "Error resolving user with id '" + idNum + "' family name '" + fName
+ 				  + "' and given name '" + gName + "'";
+ 		return cantFindUser;
+ 	}
```
Comment (ivange94) [inline api/src/main/java/org/openmrs/hl7/impl/HL7ServiceImpl.java]: Did you intend for this method to be part of the API? If you are just wrapping redundant code, then this method should be private as it is not part of the api.

File: api/src/main/java/org/openmrs/util/Security.java [hunk 2]
```diff
@@ -70,6 +70,17 @@ public static boolean hashMatches(String hashedPassword, String passwordToHash)
...
 	
 	/**
+ 	 * This method is called when password cannot be encoded
+ 	 * @param algo algorithm used for encoding
+ 	 * @return the error message string with algorithm type used
+ 	 */
+ 	public static String passwordEncodeFail(String algo) {
+ 	  String errorMessage = "Can't encode password because the given algorithm: " + algo + " was not found! (fail)";
+ 		return errorMessage;
+ 	}
```
Comment (ivange94) [inline api/src/main/java/org/openmrs/util/Security.java]: This too. Shouldn't this be a private method?

Comment (collinsnji) [inline api/src/main/java/org/openmrs/util/Security.java]: @dkayiwa, @ivange94 My local builds are passing and I can provide screenshots if you want have no issue. So I'm a little bit confused on why the travis build is failing

**Comment by collinsnji** [comment]
@dkayiwa, @ivange94 I've fixed all issues and the build now runs perfect :100:  :smiley:  If it's OK, can you please review the task so that I can move on? Thanks :smile:

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java [hunk 3]
```diff
@@ -474,7 +475,8 @@ public PatientIdentifierType getPatientIdentifierType(Integer patientIdentifierT
 
 	private String getDuplicatePatientsSQLString(List<String> attributes) {
 		String outerSelect = "select distinct t1.patient_id from patient t1 ";
-
+		String t5 = " = t5.";
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java]: Shouldn't the above even be final? :)

File: api/src/main/java/org/openmrs/hl7/impl/HL7ServiceImpl.java [hunk 4]
```diff
@@ -393,6 +393,18 @@ public HL7InError getHL7InErrorByUuid(String uuid) throws APIException {
 		return dao.getHL7InErrorByUuid(uuid);
 	}
 	
+	/** 
+ 	 * @param idNum id number
+ 	 * @param fName family name
+ 	 * @param gName given name
+ 	 * @return error string. User can not be resolveUserId
+ 	 */
+ 	private String errorFindingUser(String idNum, String fName, String gName) {
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/hl7/impl/HL7ServiceImpl.java]: I would rename this to start like get.......

File: api/src/main/java/org/openmrs/hl7/impl/HL7ServiceImpl.java [hunk 5]
```diff
@@ -393,6 +393,18 @@ public HL7InError getHL7InErrorByUuid(String uuid) throws APIException {
 		return dao.getHL7InErrorByUuid(uuid);
 	}
 	
+	/**
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/hl7/impl/HL7ServiceImpl.java]: Can you add a description for the method?

**Comment by dkayiwa** [comment]
@collinsnji next time, to make life easier, you would create a different pull request for simply removing unused imports. Then you would create another for those new methods you introduced. :)

**Comment by collinsnji** [comment]
@dkayiwa OK. The thing is I did not find any Wiki page on how method names should be assigned. 
But thanks, I'll take that into consideration next time :)

**Comment by dkayiwa** [comment]
@collinsnji address those comments and i will merge. :)

**Comment by collinsnji** [comment]
OK @dkayiwa. When I'm done is it ok to notify you on IRC?

File: api/src/main/java/org/openmrs/hl7/impl/HL7ServiceImpl.java [hunk 6]
```diff
@@ -393,6 +393,18 @@ public HL7InError getHL7InErrorByUuid(String uuid) throws APIException {
 		return dao.getHL7InErrorByUuid(uuid);
 	}
 	
+	/** Log an error when the user with the details idNumber, familyName and givenName cannot be found
+ 	 * @param idNum id number
+ 	 * @param fName family name
+ 	 * @param gName given name
+ 	 * @return error string. User can not be resolveUserId
+ 	 */
+ 	private String getErrorFindingUserMessage(String idNum, String fName, String gName) {
```
Comment (collinsnji) [inline api/src/main/java/org/openmrs/hl7/impl/HL7ServiceImpl.java]: Lol, :smile: ok let me change it

File: api/src/main/java/org/openmrs/util/Security.java [hunk 7]
```diff
@@ -70,6 +70,17 @@ public static boolean hashMatches(String hashedPassword, String passwordToHash)
 	}
 	
 	/**
+	 * Log an error with the type of algorithm used for password encoding, if the password encode fails
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/util/Security.java]: Are you sure this method logs an error? :)

File: api/src/main/java/org/openmrs/hl7/impl/HL7ServiceImpl.java [hunk 8]
```diff
@@ -393,6 +393,18 @@ public HL7InError getHL7InErrorByUuid(String uuid) throws APIException {
 		return dao.getHL7InErrorByUuid(uuid);
 	}
 	
+	/** Log an error when the user with the details idNumber, familyName and givenName cannot be found
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/hl7/impl/HL7ServiceImpl.java]: Does this method log any error? Look at it from the perspective of someone who may want to use the method. What will the method do for them?

Comment (dkayiwa) [inline api/src/main/java/org/openmrs/hl7/impl/HL7ServiceImpl.java]: You could take a look at classes in the JDK to see how methods are described in the javadoc.

Comment (collinsnji) [inline api/src/main/java/org/openmrs/hl7/impl/HL7ServiceImpl.java]: Show an error message? :grin:

**Comment by dkayiwa** [comment]
Thanks @collinsnji 👍

**Pull request merged**